include/psp2/kernel/threadmgr.h

```diff
@@ -1135,6 +1135,34 @@ int sceKernelDeleteLwCond(SceKernelLwCondWork *pWork);
 int sceKernelSignalLwCond(SceKernelLwCondWork *pWork);
 int sceKernelWaitLwCond(SceKernelLwCondWork *pWork,  unsigned int *pTimeout);
 
+typedef struct SceKernelWaitSignalResult {
```

Comment:
Where is this struct used?

Reply:
Right below in SceKernelWaitSignalOptParam.
It was used like below in sceult.
```c
SceKernelWaitSignalResult res;
SceKernelWaitSignalOptParam param;
param.result = &res;
sceKernelWaitSignal(0, 0, 0, &param); 
if (res.dret != 0) {
   // fail
}
```

Reply:
an another question, why return field has `d`?

Reply:
I don't know. It's from sceult assert message. (The message was "dret == 0" iirc) Maybe it's unix convention. I'm seeing some unix people naming error variable as dret.

Reply:
imo, it wouldn't be unix convention. just meaning of the decimal or something.
but i see that.

Reply:
I looked into this again. It turned out the fourth argument doesn't exist. It was just a random coincidence that an important struct used by libult was inside fourth register and not cleared.